data/data/bootstrap/baremetal/files/usr/local/bin/master-bmh-update.sh

```diff
@@ -6,29 +6,21 @@ export KUBECONFIG=/opt/openshift/auth/kubeconfig-loopback
 
 # Wait till the baremetalhosts are populated
 until oc get baremetalhosts -n openshift-machine-api; do
-   echo Waiting for BareMetalHosts to appear...
+   echo Waiting for BareMetalHosts CR to appear...
    sleep 20
 done
 
-AUTH_DIR=/opt/metal3/auth
-set +x
-ironic_url="$(printf 'http://%s:%s@localhost:6385/v1' "$(cat "${AUTH_DIR}/ironic/username")" "$(cat "${AUTH_DIR}/ironic/password")")"
-
-# Wait for a master to appear.
-while [ "$(curl -s "${ironic_url}/nodes" | jq '.nodes[] | .uuid' | wc -l)" -lt 1 ]; do
-    echo waiting for a master node to show up
+while [ "$(oc get bmh -n openshift-machine-api -o json | jq '.items[].metadata.name' | wc -l)" -lt 1  ]; do
+    echo "Waiting for bmh"
     sleep 20
 done
 
-# Wait for the nodes to become active after introspection.
-# Probably don't need this but I want to be 100% sure.
-while curl -s "${ironic_url}/nodes" | jq '.nodes[] | .provision_state' | grep -v active; do
-    echo Waiting for nodes to become active
+while [ "$(oc get bmh -n openshift-machine-api -o json | jq '.items[].status.provisioning.state' | grep -v \"\" | grep -v provisioned | wc -l)" -gt 0  ]; do
```

Comment:
This is a place where it would be convenient to have a label on the control plane hosts so we can select for them instead of doing `grep -v \"\"` to filter out the workers.
Technically this is prone to races because before any of the control plane hosts have started registration, none of them will match this filter. So there is a window after the BMHs are created but before BMO has reconciled any of them where we could skip straight to shutting everything down again.

Reply:
Done